### PR TITLE
Add nullable annotations to Healthchecks

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/HealthChecks/Abstractions/src/HealthCheckContext.cs [hunk 1]
```diff
@@ -8,6 +8,6 @@ public sealed class HealthCheckContext
         /// <summary>
         /// Gets or sets the <see cref="HealthCheckRegistration"/> of the currently executing <see cref="IHealthCheck"/>.
         /// </summary>
-        public HealthCheckRegistration Registration { get; set; }
+        public HealthCheckRegistration Registration { get; set; } = default!;
```
Comment (JamesNK) [inline src/HealthChecks/Abstractions/src/HealthCheckContext.cs:11]: Was this only nullable for testing? Should add comment why it has been forced to non-nullable

File: src/HealthChecks/HealthChecks/test/DefaultHealthCheckServiceTest.cs [hunk 2]
```diff
@@ -243,7 +243,7 @@ public async Task CheckHealthAsync_SetsRegistrationForEachCheck()
         public async Task CheckHealthAsync_Cancellation_CanPropagate()
         {
             // Arrange
-            var insideCheck = new TaskCompletionSource<object>();
+            var insideCheck = new TaskCompletionSource<object?>();
```
Comment (JamesNK) [inline src/HealthChecks/HealthChecks/test/DefaultHealthCheckServiceTest.cs:246]: Non generic TaskCompletionSource is merged in runtime. We should have it soon 🙏

File: src/HealthChecks/Abstractions/src/HealthCheckContext.cs (hunk 1 above)
Comment (pranavkm) [reply src/HealthChecks/Abstractions/src/HealthCheckContext.cs:11]: Yeah, the calling code assumes this is non-null.

Comment (pranavkm) [reply src/HealthChecks/Abstractions/src/HealthCheckContext.cs:12]: Done

**Review by BrennanConroy — APPROVED** [review]
We no longer bringing these to API review?

**Comment by pranavkm** [comment]
@BrennanConroy the suggestion was to review these together rather than individually. I'll set up a separate review once we've had more of these projects updated.

**Pull request merged**